### PR TITLE
Add Token Endpoint and add Authorization Discovery

### PR DESCRIPTION
**Pull request by dshanske**
This adds the discovery stuff from previous with the token endpoint items for today. It makes a bunch of changes recommended by @aaronpk as well.

Will likely hide the option to use an external token endpoint in a future version, as it seems that most people would want the built-in.

### CODE REVIEW

**Comment by dshanske** [comment]
Because @aaronpk had an issue with using localhost in testing, I added the DEBUG file @pfefferle created for webmentions when he had the same problem. Really looking forward to getting this in.

File: includes/class-indieauth-authenticate.php [hunk 1]
```diff
@@ -81,50 +100,78 @@ public function verify_access_token( $token ) {
...
+				'me'            => rawurlencode( $me ),
+				'redirect_uri'  => rawurlencode( $redirect_uri ),
+				'client_id'     => rawurlencode( home_url() ),
+				'state'         => $state,
+				'response_type' => 'id',
 			)
 		);
 		// redirect to authentication endpoint
-		wp_redirect( get_option( 'indieauth_authorization_endpoint' ) . '?' . $query );
+		wp_redirect( $authorization_endpoint . '?' . $query );
```
Comment (pfefferle) [inline includes/class-indieauth-authenticate.php]: you should use a more solid method to add the query to the authorization endpoint. If hoster does not support `mod_rewrite`, `$authorization_endpoint` might already has query params.

File: includes/class-indieauth-token-endpoint.php [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+ */
+
+class IndieAuth_Token_Endpoint {
+	public function __construct() {
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+
+	}
+
+
+		/**
```
Comment (pfefferle) [inline includes/class-indieauth-token-endpoint.php]: indents!

File: includes/class-indieauth-token-ui.php [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	}
+
+
+	/**
+	 * Generate Options Form.
+	 *
+	 * @access public
+	 */
+	public function options_form() {
+		echo '<div class="wrap">';
```
Comment (pfefferle) [inline includes/class-indieauth-token-ui.php]: perhaps use a template, or at least use a more template style output (don't echo HTML tags)

File: functions.php [hunk 4]
```diff
@@ -0,0 +1,155 @@
+<?php
```
Comment (pfefferle) [inline functions.php]: remove this file, because it is a duplicate of `includes/functions.php`

**Comment by dshanske** [comment]
@pfefferle I removed the accidental adding of a duplicate functions file, changed the build_query to add_query_args which is more redundant, and moved the form to a template. Should do the trick.

**Pull request merged**